docs/explanation/endianness.rst

```diff
@@ -2,30 +2,30 @@
 Order of qubits in Qiskit
 #########################
 
-While most physics textbooks represent an :math:`n`-qubit system as the tensor product :math:`Q_0\otimes Q_1 \otimes ... \otimes Q_{n-1}`, where :math:`Q_j` is the :math:`j^{\mathrm{th}}` qubit, Qiskit uses the inverse order, that is, :math:`Q_{n-1}\otimes ... \otimes Q_1 \otimes Q_{0}`. As explained in `this video <https://www.youtube.com/watch?v=EiqHj3_Avps>`_ from `Qiskit's YouTube channel <https://www.youtube.com/@qiskit>`_, this is done to follow the convention in classical computing, in which the :math:`n^{\mathrm{th}}` bit or most significant bit (MSB) is placed on the left (with index 0) while the least significant bit (LSB) is placed on the right (index :math:`n-1`). This ordering convention is called little-endian while the one from the physics textbooks is called big-endian.
+While most physics textbooks represent an :math:`n`-qubit system as the tensor product :math:`Q_0\otimes Q_1 \otimes ... \otimes Q_{n-1}`, where :math:`Q_j` is the :math:`j^{\mathrm{th}}` qubit, Qiskit uses the inverse order, that is, :math:`Q_{n-1}\otimes \dotsb \otimes Q_1 \otimes Q_{0}`. As explained in `this video <https://www.youtube.com/watch?v=EiqHj3_Avps>`_ from `Qiskit's YouTube channel <https://www.youtube.com/@qiskit>`_, this is done to follow the convention in classical computing, in which the :math:`n^{\mathrm{th}}` bit or most significant bit (MSB) is placed on the left while the least significant bit (LSB) is placed on the right. This ordering convention is called little-endian while the one from the physics textbooks is called big-endian.
```

Comment:
I think that the explanation here in the PR is ok as is. I don't think there is a perfect way to explain this concept, what is important for beginners is to understand that a) most physics textbooks do it one way, b) qiskit does it a different way, and c) some minimal explanation for why qiskit is different. Also if we want I think it is perfectly ok for us to say we do it this way because it is convenient programmatically haha

Reply:
Perhaps I'm being a bit too passive/polite in my comment at the top.  This description is at best misleading, at worst incorrect about things it's attempting to clear up.  I don't mind exactly how much information about the intuition you feel is necessary.  I'd argue that our "explainers" shouldn't _only_ be targetted at beginners, though, and intuition is a key part of teaching somebody something.

Specifically:

> While most physics textbooks represent a $n$-qubit state as a tensor product $Q_0\otimes Q_1 \otimes \dotsb \otimes Q_{n-1}$, where $Q_j$ is the $j^{\mathrm{th}}$ qubit, Qiskit uses the inverse order, that is, $Q_{n-1}\otimes \dotsb \otimes Q_1 \otimes Q_{0}$.

No part of this sentence defines a "most significant bit".  This is my point above: the explanation as written uses that terminology, but it's confused about what makes something a most significant bit, and it confuses the concepts of "which bit is the most significant?" and "which bit do I call 'bit 0'?" which are separate questions.

> in which the $n^{\mathrm{th}}$ bit or most significant bit (MSB) is placed on the left while the least significant bit (LSB) is placed on the right.

This is the crux of my previous sentence.  The order you place the MSB and LSB is the endianness of a representation.  Putting the MSB of a classical integer on the left in written text (little-endian) is a near universal convention.  The question of "do you call the MSB or the LSB 'bit 0'?" is a different question, as is "what makes a _qubit_ the most/least significant?".

If you don't want to talk about the intuition at all, then I would leave out _all_ discussion of most- and least-significant bits, because the link to MSB/LSB _is_ the intuition part.  Doing that could look something like

> Physics textbooks often number the qubits in a multi-qubit state starting at one from the left, such as $\lvert x_1x_2x_3\rangle$.  By contrast, Qiskit labels the qubits in a multi-qubit state starting at zero from the right, such as $\lvert q_2q_1q_0\rangle$.
>
> This affects several parts of the library.  For example in Qiskit, the Pauli string `"XYZ"` represents a $Z$ operation on qubit 0, a $Y$ on qubit 1, and an $X$ on qubit 2.  Similarly, the representation of $CX$ controlled on qubit 0 targetting qubit 1 is written $1_1\otimes\lvert0\rangle\langle0\rvert_0 + X_1\otimes\lvert1\rangle\langle1\rvert_0$, which has the matrix representation [...].

Reply:
Fwiw: I'm not going to edit the text of the above to hide what I said, but I regret saying it like that and I'm sorry for it.  I've ended up being rude about Junye's/Guillermo's work, and I'm not best pleased with the rest of the tone of my response either.

I do think that the current explanation ought to be reworked, but the way I said it in the second response wasn't the right of doing that.